frontend/src/hooks/useDocuments.js

```diff
@@ -0,0 +1,92 @@
+import { apiRoutes } from '@/constants/routes'
+import { useApiService } from '@/services/useApiService'
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query'
+
+export const useDocuments = (parentType, parentID, options) => {
+  const client = useApiService()
+
+  return useQuery({
+    queryKey: ['documents', parentType, parentID],
+    queryFn: async () => {
+      const path = apiRoutes.getDocuments
+        .replace(':parentID', parentID)
+        .replace(':parentType', parentType)
+
+      const res = await client.get(path)
+      return res.data
+    },
+    ...options
+  })
+}
+
+export const useUploadDocument = (parentType, parentID, options) => {
+  const client = useApiService()
+  const queryClient = useQueryClient()
+  const path = apiRoutes.getDocuments
+    .replace(':parentID', parentID)
+    .replace(':parentType', parentType)
+
+  return useMutation({
+    ...options,
+    mutationFn: async (file) => {
+      const formData = new FormData()
+      formData.append('file', file)
+      formData.append('filename', file.name)
+
+      return await client.post(path, formData, {
+        headers: {
+          'Content-Type': 'multipart/form-data'
+        }
+      })
+    },
+    onSuccess: () => {
+      queryClient.invalidateQueries(['documents', parentType, parentID])
+    }
+  })
+}
+
+export const useDeleteDocument = (parentType, parentID, options) => {
```

Comment:
getting an error when trying to delete: "DELETE /api/documents/compliance_report/1/1 HTTP/1.1" 405 Method Not Allowed"

I noticed there is a delete file route in compliance reporting now so is this hook pointing at the wrong delete endpoint?

@router.delete(
    "/{report_id}/files/{document_id}",
)

Reply:
Fixed! Sorry about that, forgot to move the delete method to the document controller. MR is updated.